tensorflow_lite_support/ios/test/task/audio/core/audio_record/TFLAudioRecordTests.m

```diff
@@ -0,0 +1,296 @@
+/* Copyright 2021 The TensorFlow Authors. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ ==============================================================================*/
+#import <AVFoundation/AVFoundation.h>
+#import <XCTest/XCTest.h>
+
+#import "tensorflow_lite_support/ios/sources/TFLCommon.h"
+#import "tensorflow_lite_support/ios/test/task/audio/core/audio_record/utils/sources/AVAudioPCMBuffer+Utils.h"
+
+#import "tensorflow_lite_support/ios/task/audio/core/audio_record/sources/TFLAudioRecord.h"
+
+#define VerifyError(error, expectedErrorDomain, expectedErrorCode, expectedLocalizedDescription) \
+  XCTAssertEqual(error.domain, expectedErrorDomain);                                             \
+  XCTAssertEqual(error.code, expectedErrorCode);                                                 \
+  XCTAssertEqualObjects(error.localizedDescription, expectedLocalizedDescription);
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface TFLAudioRecordTests : XCTestCase
+@property(nonatomic) AVAudioFormat *audioEngineFormat;
+@end
+
+// This category of TFLAudioRecord is private to the current test file.
+// This is needed in order to expose the method to load the audio record buffer 
+// without calling: -[TFLAudioRecord startRecordingWithError:].
+// This is needed to avoid exposing this method which isn't useful to the consumers
```

Comment:
@khanhlvg Swift tests cannot be added for loading and readAtOffset since there is no similar workaround to expose this private method to tests without exposing it in the framework too.

Reply:
That's fine. We only need to test the internal logic of TFLAudioRecord so having only ObjC tests is enough.